### PR TITLE
Replace jest with vitest

### PR DESCRIPTION
**Pull request by renahashimi**
**Summary of Changes for #118 **
- Removed Jest: Uninstalled Jest and replaced it with Vitest for testing.
- Added Coverage: Integrated coverage reporting with Vitest.
- Updated Tests:
-- Added new tests for the sum function, including:
-- Positive and negative number additions.
-- String inputs resulting in valid sums or NaN.
- Updated .gitignore: Added coverage/ to exclude generated coverage reports from version control.

### CODE REVIEW

File: src/js/test/sum.js
Comment (jonhenrikaavitsland) [inline src/js/test/sum.js:1]: This file was not part of the task, but i accept it as a upcoming task will override this change.

**Review by jonhenrikaavitsland — CHANGES_REQUESTED** [review]
@renahashimi 
You haven't removed the config file for jest. 
Please fix this and push changes.

File: src/js/test/sum.js
Comment (renahashimi) [reply src/js/test/sum.js:1]: @jonhenrikaavitsland 
Sorry about that :). The file was already there , so I added more test to see if the testing works. I will remove the "config file for jest" later today ;)

Comment (jonhenrikaavitsland) [reply src/js/test/sum.js:1]: ok that is fine. 
writing tests can be a different task :)

**Review by jonhenrikaavitsland — APPROVED** [review]
Thank you. I will merge your PR now

**Pull request merged**